stable/launcher/powershell-wrapper

```diff
@@ -45,11 +52,19 @@ TestDistro(){
     # Tests for unsupported distrobutions go here
     if [ "${DIST}" = "fedora" -a "${REV}" = "28" ] ; then
         ReportUnsupported -D $DIST -R $REV
-        # Workaround for https://github.com/powershell/powershell/issues/6132
-        export TERM=xterm
+        echoerr "Use 'powershell-preview' package for support."
+        WorkaroundTerminalIssue
+    elif [ "${DIST}" = "manjaro" ] ; then
+        ReportUnsupported -D $DIST -R $REV
+        WorkaroundTerminalIssue
     fi
 }
 
+WorkaroundTerminalIssue(){
```

Comment:
Something I'm concerned about is that as time goes on, and Snap supports more distros we run into more issues and have to add other workarounds like this making this file large.

Ideally, I wouldn't want a bunch of functions in this file, but pulled out to some other file if possible like a "workarounds" script.

Or a workaround script per distro with the workarounds per revision.

Since this is the only workaround so far... I wont block on this, but it's something think about for the future... unless you want to address it now :)

Reply:
There is already a separate script for 6.1 where DotNet has fixed this issue.  The distro detection needs to happen in one file.  For PowerShell installation, we follow the model you suggest but on the family basis.  I think we should handle it when we get big enough that it matters.  So far, this issue seems to only occur on RedHat variants (Manjaro seem to be more related to RedHat that Debian.)  If I had better detection of the issue itself, your suggestion might be useful.